packages/fast-components-react-msft/src/pivot/pivot.tsx

```diff
@@ -0,0 +1,240 @@
+        }
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLButtonElement | HTMLAnchorElement> {
+        return (
+            <BaseTabs
+                {...this.unhandledProps()}
+                ref={this.tabsRef}
+                managedClasses={this.generatePivotClassNames()}
+                activeId={this.state.activeId}
+                onUpdate={this.handleTabsUpdate}
+                items={this.props.items}
+                label={this.props.label}
+            >
+                <span
+                    style={{ transform: `translateX(${toPx(this.state.offsetX)})` }}
+                    className={get(
+                        this.props,
+                        "managedClasses.pivot_activeIndicator",
+                        ""
+                    )}
+                />
+            </BaseTabs>
+        );
+    }
+
+    /**
+     * Returns tabs managedclasses with new carousel-specific JSS
+     */
+    protected generatePivotClassNames(): TabsClassNameContract {
+        return {
+            tabs: get(this.props, "managedClasses.pivot", ""),
+            tabs_tabPanels: this.generateTabPanelsClassNames(),
+            tabs_tabList: get(this.props, "managedClasses.pivot_tabList", ""),
+            tabs_tabContent: get(this.props, "managedClasses.pivot_tabContent", ""),
+            tabs_tabPanelContent: get(
+                this.props,
+                "managedClasses.pivot_tabPanelContent",
+                ""
+            ),
+            tab: get(this.props, "managedClasses.pivot_tab", ""),
+            tab__active: get(this.props, "managedClasses.pivot_tab__active", ""),
+            tabPanel: get(this.props, "managedClasses.pivot_tabPanel", ""),
+            tabPanel__hidden: get(
+                this.props,
+                "managedClasses.pivot_tabPanel__hidden",
+                ""
+            ),
+        };
+    }
+
+    private handleTabsUpdate = (activeTabId: string): void => {
+        this.setState({
+            activeId: activeTabId,
+        });
+    };
+
+    private isSelected(element: HTMLElement): boolean {
+        return element.className.includes("active") === true;
+    }
+
+    private updateTabPanelIndex(): void {
+        if (canUseDOM() && this.tabsRef.current && Array.isArray(this.props.items)) {
+            const tabElement: HTMLElement = ReactDOM.findDOMNode(
+                this.tabsRef.current
+            ) as HTMLElement;
+
+            const mytabsArray: HTMLElement[] = Array.prototype.slice.call(
+                tabElement.querySelectorAll("[role='tab']")
+            );
+            this.setState({
+                tabPanelIndex: mytabsArray.findIndex(this.isSelected),
+            });
+        }
+    }
+
+    private generateTabPanelsClassNames(): string {
+        let className: string = get(this.props, "managedClasses.pivot_tabPanels", "");
+        if (this.state.tabPanelIndex === this.prevTabPanelIndex) {

```

Comment:
What is the purpose for assigning `className` to itself? I'm 98% sure you can refactor this to avoid that check